src/test/mediapipe_validation_test.cpp

```diff
@@ -0,0 +1,169 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <chrono>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../grpcservermodule.hpp"
+#include "../kfs_frontend/kfs_grpc_inference_service.hpp"
+#include "../mediapipe_internal/mediapipegraphexecutor.hpp"
+#include "../precision.hpp"
+#include "../server.hpp"
+#include "test_utils.hpp"
+
+using namespace ovms;
+
+class MediapipeValidationTest : public ::testing::Test {
+protected:
+    ovms::Server& server = ovms::Server::instance();
+    const ovms::Module* grpcModule;
+    KFSInferenceServiceImpl* impl;
+    ::KFSRequest request;
+    ::KFSResponse response;
+
+    const Precision precision = Precision::FP32;
+    std::unique_ptr<std::thread> t;
+    std::string port = "9178";
+    void SetUpServer(const char* configPath) {
+        server.setShutdownRequest(0);
+        randomizePort(this->port);
+        char* argv[] = {(char*)"ovms",
+            (char*)"--config_path",
+            (char*)configPath,
+            (char*)"--port",
+            (char*)port.c_str()};
+        int argc = 5;
+        t.reset(new std::thread([&argc, &argv, this]() {
+            EXPECT_EQ(EXIT_SUCCESS, server.start(argc, argv));
+        }));
+        auto start = std::chrono::high_resolution_clock::now();
+        while ((server.getModuleState(SERVABLE_MANAGER_MODULE_NAME) != ovms::ModuleState::INITIALIZED) &&
+               (!server.isReady()) &&
+               (std::chrono::duration_cast<std::chrono::seconds>(std::chrono::high_resolution_clock::now() - start).count() < 5)) {
+        }
+        grpcModule = server.getModule(ovms::GRPC_SERVER_MODULE_NAME);
+        impl = &dynamic_cast<const ovms::GRPCServerModule*>(grpcModule)->getKFSGrpcImpl();
+    }
+    void prepareSingleInput() {
+        request.Clear();
+        response.Clear();
+        inputs_info_t inputsMeta{{"in", {DUMMY_MODEL_SHAPE, precision}}};
+        preparePredictRequest(request, inputsMeta);
+        request.mutable_model_name()->assign("mediapipeDummyADAPTFULL");
+    }
+    void prepareDoubleInput() {
+        request.Clear();
+        response.Clear();
+        inputs_info_t inputsMeta{
+            {"in1", {DUMMY_MODEL_SHAPE, precision}},
+            {"in2", {DUMMY_MODEL_SHAPE, precision}}};
+        std::vector<float> requestData1{0., 0., 0., 0., 0., 0., 0, 0., 0., 0};
+        std::vector<float> requestData2{0., 0., 0., 0., 0., 0., 0, 0., 0., 0};
+        preparePredictRequest(request, inputsMeta, requestData1);
+        request.mutable_model_name()->assign("mediapipeAddADAPTFULL");
+    }
+    void SetUp() override {
+        SetUpServer("/ovms/src/test/mediapipe/config_mediapipe_all_graphs_adapter_full.json");
```

Comment:
Since we setup server the same way for each test it makes more sense to use SetUpTestSuite to bear the cost of this only once.

Reply:
done